src/graph/graph_apis.cc

```diff
@@ -383,58 +434,45 @@ DGL_REGISTER_GLOBAL("graph_index._CAPI_DGLDisjointUnion")
+    const ImmutableGraph *im_gr = dynamic_cast<const ImmutableGraph *>(ptr);
+    const Graph *gr = dynamic_cast<const Graph *>(ptr);
+    if (gr) {
+      DGLDisjointUnion<Graph>(inhandles, list_size, rv);
+    }
+    if (im_gr) {
+      DGLDisjointUnion<ImmutableGraph>(inhandles, list_size, rv);
     }
-    Graph* gptr = new Graph();
-    *gptr = GraphOp::DisjointUnion(std::move(graphs));
-    GraphHandle ghandle = gptr;
-    *rv = ghandle;
   });
 
 DGL_REGISTER_GLOBAL("graph_index._CAPI_DGLDisjointPartitionByNum")
 .set_body([] (DGLArgs args, DGLRetValue* rv) {
     GraphHandle ghandle = args[0];
     const GraphInterface *ptr = static_cast<const GraphInterface *>(ghandle);
     const Graph* gptr = dynamic_cast<const Graph*>(ptr);
-    CHECK(gptr) << "_CAPI_DGLDisjointPartitionByNum isn't implemented in immutable graph";
-    int64_t num = args[1];
-    std::vector<Graph>&& rst = GraphOp::DisjointPartitionByNum(gptr, num);
-    // return the pointer array as an integer array
-    const int64_t len = rst.size();
-    NDArray ptr_array = NDArray::Empty({len}, DLDataType{kDLInt, 64, 1}, DLContext{kDLCPU, 0});
-    int64_t* ptr_array_data = static_cast<int64_t*>(ptr_array->data);
-    for (size_t i = 0; i < rst.size(); ++i) {
-      Graph* ptr = new Graph();
-      *ptr = std::move(rst[i]);
-      ptr_array_data[i] = reinterpret_cast<std::intptr_t>(ptr);
+    const ImmutableGraph* im_gptr = dynamic_cast<const ImmutableGraph*>(ptr);
+    if (gptr) {
+      DGLDisjointPartitionByNum(gptr, args, rv);
+    }
+    if (im_gptr){
+      DGLDisjointPartitionByNum(im_gptr, args, rv);
     }
-    *rv = ptr_array;
   });
 
 DGL_REGISTER_GLOBAL("graph_index._CAPI_DGLDisjointPartitionBySizes")
 .set_body([] (DGLArgs args, DGLRetValue* rv) {
     GraphHandle ghandle = args[0];
+    const IdArray sizes = IdArray::FromDLPack(CreateTmpDLManagedTensor(args[1]));
     const GraphInterface *ptr = static_cast<const GraphInterface *>(ghandle);
     const Graph* gptr = dynamic_cast<const Graph*>(ptr);
-    CHECK(gptr) << "_CAPI_DGLDisjointPartitionBySizes isn't implemented in immutable graph";
-    const IdArray sizes = IdArray::FromDLPack(CreateTmpDLManagedTensor(args[1]));
-    std::vector<Graph>&& rst = GraphOp::DisjointPartitionBySizes(gptr, sizes);
-    // return the pointer array as an integer array
-    const int64_t len = rst.size();
-    NDArray ptr_array = NDArray::Empty({len}, DLDataType{kDLInt, 64, 1}, DLContext{kDLCPU, 0});
-    int64_t* ptr_array_data = static_cast<int64_t*>(ptr_array->data);
-    for (size_t i = 0; i < rst.size(); ++i) {
-      Graph* ptr = new Graph();
-      *ptr = std::move(rst[i]);
-      ptr_array_data[i] = reinterpret_cast<std::intptr_t>(ptr);
+    const ImmutableGraph* im_gptr = dynamic_cast<const ImmutableGraph*>(ptr);
+    if (gptr) {
+      DGLDisjointPartitionBySizes(gptr, sizes, rv);
     }
-    *rv = ptr_array;
-  });
+    if (im_gptr){
+      DGLDisjointPartitionBySizes(im_gptr, sizes, rv);
+    }
```

Comment:
same above

Reply:
Fixed. (Don't know why it isn't showed here)